File: 0443-string-compression/0443-string-compression.cs
public class Solution {
    public int Compress(char[] chars) {
        int i=0;
        int index=0;
        int n =chars.Length;
        
        while (i<n){
            char currentchar = chars[i];
            int count =0;
            while(i<n&& chars[i]==currentchar){
                count++;
                i++;
            }
            chars[index]= currentchar;
            index++;
            if (count>1){
                string countstr = count.ToString();
                foreach ( char ch in countstr){
                    chars[index]=ch;
                    index++;
                }
            }
        }
        return index;
        
    }
}

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int read =0;
        int write =0;
        while (read<chars.length){
            int count=0;
            char ch = chars[read];
            while (read<chars.length&& chars[read]==ch){
                count ++;
                read ++;
            }
            chars[write]=ch;
            write++;
            if (count >1){
                String s = String.valueOf(count);
                for ( int i=0;i<s.length();i++){
                    chars[write]=s.charAt(i);
                    write++;
                }


            }
        }
        return write ;
        
    }
}